### PR TITLE
raise more specific error if dict is appended to frame wit…

### LINKED ISSUE #30871: Improve error message for DataFrame.append(<dict-like>)
#### Code Sample, a copy-pastable example if possible

```python
>>> import pandas as pd
>>>
>>> pd.__version__
'0.26.0.dev0+1731.g3ddd495e4'
>>>
>>> df = pd.DataFrame({"Name": ["Alice"], "Gender": ["F"]})
>>> df
    Name Gender
0  Alice      F
>>>
>>> df.append({"Name": "Bob", "Gender": "M"})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\simon\pandas\pandas\core\frame.py", line 7013, in append
    "Can only append a Series if ignore_index=True "
TypeError: Can only append a Series if ignore_index=True or if the Series has a name

>>>

```
#### Problem description
dict-like is valid type for `other` parameter of DataFrame.append

#### Expected Output

ValueError: Can only append a dict-like if ignore_index=True

#### Output of ``pd.show_versions()``

<details>

[paste the output of ``pd.show_versions()`` here below this line]

</details>

### PR DESCRIPTION
**Pull request by MarcoGorelli**
…hout ignore_index

- [x] closes #30871 
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by simonjayhawkins — APPROVED** [review]
Thanks @MarcoGorelli for the PR. lgtm. although I _think_ that a note in the other section of the whatsnew to say the message has been changed might be warranted. see what others think.

File: doc/source/whatsnew/v1.0.0.rst [hunk 1]
```diff
@@ -1170,6 +1170,8 @@ Other
 - Bug where :meth:`DataFrame.itertuples` would incorrectly determine whether or not namedtuples could be used for dataframes of 255 columns (:issue:`28282`)
 - Handle nested NumPy ``object`` arrays in :func:`testing.assert_series_equal` for ExtensionArray implementations (:issue:`30841`)
 - Bug in :class:`Index` constructor incorrectly allowing 2-dimensional input arrays (:issue:`13601`, :issue:`27125`)
+- Appending a dictionary to a :class:`DataFrame` without passing ``ignore_index=True`` will raise ``TypeError: Can only append a dict if ignore_index=True``
+  instead of ``TypeError: Can only append a Series if ignore_index=True or if the Series has a name`` (:issue`30871`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v1.0.0.rst]: This will need to go in doc/source/whatsnew/v1.1.0.rst once #30907 is merged

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py [hunk 2]
```diff
@@ -7023,6 +7023,8 @@ def append(
         """
         if isinstance(other, (Series, dict)):
             if isinstance(other, dict):
+                if not ignore_index:
```
Comment (WillAyd) [inline pandas/core/frame.py]: Rather than adding another branch here can you just use an f-string and do `f"Can only append a {type(other)}...`

**Review by WillAyd — CHANGES_REQUESTED** [review]
The 1.1 file should be available now to move whatsnew

File: pandas/core/frame.py (hunk 2 above)
Comment (MarcoGorelli) [reply pandas/core/frame.py]: Could do, but then I'll need to add another branch later as the message for Series has an extra part compared to that for dict: "or if the Series has a name"

**Review by simonjayhawkins — APPROVED** [review]

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @MarcoGorelli

**Pull request merged**